oak_runtime/src/lib.rs

```diff
@@ -733,25 +741,34 @@ impl Runtime {
         &self,
         node_id: NodeId,
         source_label: &Label,
+        use_privilege: OperationPrivilege,
     ) -> Result<(), OakStatus> {
-        // When reading from a Channel, we downgrade the Label of the Channel from which the Node is
-        // reading: the thing being downgraded is the data being read into the Node (protected by
-        // the Channel Label).
-        let downgraded_source_label = self.get_node_downgraded_label(node_id, source_label);
         let target_label = self.get_node_label(node_id);
         let node_name = self.get_node_name(node_id);
         trace!(
             "{:?}: original source label: {:?}?",
             node_name,
             source_label
         );
-        trace!(
-            "{:?}: downgraded source label: {:?}?",
-            node_name,
-            downgraded_source_label
-        );
+
+        let final_source_label = match use_privilege {
+            OperationPrivilege::UsePrivilege => {
+                // When reading from a Channel and using privilege, we downgrade the Label of the
+                // Channel from which the Node is reading: the thing being downgraded is the
+                // data being read into the Node (protected by the Channel Label).
+                let downgraded_source_label = self.get_node_downgraded_label(node_id, source_label);
+
+                trace!(
+                    "{:?}: downgraded source label: {:?}?",
+                    node_name,
+                    downgraded_source_label
+                );
+                downgraded_source_label
+            }
+            OperationPrivilege::None => source_label.to_owned(),
+        };
```

Comment:
Could this chunk be encapsulated into a common `get_effective_label()` helper method?  It looks like similar code is repeated a few times.

Reply:
Done.